src/Libraries/RevitServices/Persistence/ElementBinder.cs

```diff
@@ -47,6 +47,34 @@ public SerializableId(SerializationInfo info, StreamingContext context)
             StringID = (string) info.GetValue("stringID", typeof (string));
             IntID = (int)info.GetValue("intID", typeof(int));
         }
+
+        public override bool Equals(object other)
+        {
+            var sID = other as SerializableId;
+            if (sID == null)
+            {
+                return false;
+            }
+
+            if (this.IntID.Equals(sID.IntID) && this.StringID.Equals(sID.StringID))
+            {
+                return true;
+            }
+            return false;
+
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked // Overflow is fine, just wrap
+            {
+                int hash = 17;
+
+                hash = hash * 23 + StringID.GetHashCode();
+                hash = hash * 23 + IntID.ToString().GetHashCode();
```

Comment:
@Randy-Ma yes in this case equals is the method that is used, I've overridden hashcode as well though for completeness sake. If most are more familiar and comfortable with the stand xor hash function, of properties I will use that instead. I can't say one way or the other which will provide less collisions.

I will change the hash, and add a test.